packages/analytics-js-integrations/src/integrations/Heap/browser.js

```diff
@@ -41,20 +46,38 @@ class Heap {
   // DOC: https://developers.heap.io/reference#identify
 
   identify(rudderElement) {
-    const { userId, context } = rudderElement.message;
-    if (userId) {
-      window.heap.identify(userId);
+    try {
+      if (!this.isLoaded()) {
+        logger.error('Heap SDK not loaded. Skipping identify call.');
+        return;
+      }
```

Comment:
If the integration did not report ready, no tracking APIs will be called. So, this check is redundant.

Reply:
Removed it